core/src/main/java/software/amazon/awssdk/core/retry/conditions/OrClockSkewCondition.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.retry.conditions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * Composite retry condition that evaluates to true if any containing condition evaluates to true.
+ */
+@SdkPublicApi
+public class OrClockSkewCondition implements ClockSkewCondition {
+
+    private List<ClockSkewCondition> conditions = new ArrayList<>();
+
+    public OrClockSkewCondition(ClockSkewCondition... conditions) {
+        Collections.addAll(this.conditions, conditions);
+    }
+
+    /**
+     * @return True if any condition returns true. False otherwise.
+     */
+    @Override
+    public boolean isClockSkewError(Exception exception) {
```

Comment:
Can this be shouldRetry() instead of isClockSkewError()? Why do we need a new interface?

Reply:
Perhaps these methods belong on the exception itself rather than the retry policy. I wouldn't think customers would have much of a need to plug in their own logic for these conditions. So you could have something like this

```
if(e instanceof SdkServiceException && ((SdkServiceException) e).isClockSkewException())
```

If we do it this way it might be a good idea to have a base service exception specific to AWS services so we can put the implementation of these methods in there.

Reply:
+1 on putting it to exception